homeassistant/core.py

```diff
@@ -905,20 +905,23 @@ def entity_ids(self, domain_filter: Optional[str] = None) -> List[str]:
         return future.result()
 
     @callback
-    def async_entity_ids(self, domain_filter: Optional[str] = None) -> List[str]:
+    def async_entity_ids(
+        self, domain_filter: Optional[Union[str, Iterable]] = None
+    ) -> List[str]:
         """List of entity ids that are being tracked.
 
         This method must be run in the event loop.
         """
         if domain_filter is None:
             return list(self._states.keys())
 
-        domain_filter = domain_filter.lower()
+        if isinstance(domain_filter, str):
+            domain_filter = [domain_filter.lower()]
```

Comment:
Making it a tuple is faster I think
```suggestion
            domain_filter = (domain_filter.lower(),)
```

Reply:
Also since this is a common case, maybe not even wrap it and just copy the below code and execute it